File: S3/Programmation/Python/voiture/Model/VoitureRoute.py
class VoitureRoute:
    def __init__(self,id,voiture,code_route):
        self._id = id
        self._voiture = voiture
        self._code_route = code_route

    # id
    @property
    def id(self):
        return self._id

    @id.setter
    def id(self, value):
        self._id = value

    # voiture
    @property
    def voiture(self):
        return self._voiture

    @voiture.setter
    def voiture(self, value):
        self._voiture = value

    # code_route
    @property
    def code_route(self):
        return self._code_route

    @code_route.setter
    def code_route(self, value):
        self._code_route = value

    def __str__(self):
        return f"VoitureRoute(id={self.id}, voiture={self.voiture}, code_route='{self.code_route}')"

File: S3/Programmation/Voyage/src/model/VoitureRoute.java
package model;

import java.time.LocalDateTime;

public class VoitureRoute {
    private int id;
    private Voiture voiture;   // objet Voiture
    private String codeRoute;
    private Status status;     // objet Status
    private LocalDateTime duree;

    public VoitureRoute() {}

    public VoitureRoute(int id, Voiture voiture, String codeRoute, Status status, LocalDateTime duree) {
        this.id = id;
        this.voiture = voiture;
        this.codeRoute = codeRoute;
        this.status = status;
        this.duree = duree;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public Voiture getVoiture() { return voiture; }
    public void setVoiture(Voiture voiture) { this.voiture = voiture; }

    public String getCodeRoute() { return codeRoute; }
    public void setCodeRoute(String codeRoute) { this.codeRoute = codeRoute; }

    public Status getStatus() { return status; }
    public void setStatus(Status status) { this.status = status; }

    public LocalDateTime getDuree() { return duree; }
    public void setDuree(LocalDateTime duree) { this.duree = duree; }
}
